### PR TITLE
ceph: Add ASAN-enabled ceph container

### PR DESCRIPTION
**Pull request by peng225**
issue: cybozu/csa#82
Signed-off-by: Shinya Hayashi <shinya-hayashi@cybozu.co.jp>

### CODE REVIEW

**Comment by peng225** [comment]
The result of CI is as follows.

### build-ceph

#### Check TAG file

generated script
```
#!/bin/bash -eo pipefail
c="$(./tag_exists ceph)"  #### No suffix added ####
if [ "$c" = ng ]; then
    echo ceph > BUILDS
fi
```

### build-ceph-container

#### Check TAG files

generated script
```
#!/bin/sh -eo pipefail
dir=ceph                                 #### `dir` is set to ceph ####
if [ "$dir" = "" ]; then dir=ceph ; fi
image=ceph                                 #### `image` is set to ceph ####
targets=""
if [ "$targets" != "" ]; then
  for target in $targets; do break; done # get first element
  image=$image-$target
fi
suffix=                                     #### No suffix ####
c="$(./tag_exists $image $dir $suffix)"
```

### build-ceph-asan

#### Build Ceph packages with ASAN enabled

generated script
```
# set WITH_ASAN flag to ON
ceph/deb.sh 16.2.7 ON
```

console output
```
Resolving deltas: 100% (242/242)
Resolving deltas: 100% (242/242), completed with 214 local objects.
From ssh://github.com/facebook/zstd
 * branch            b706286adbba780006a47ef92df0ad7a785666b6 -> FETCH_HEAD
Submodule path 'src/zstd': checked out 'b706286adbba780006a47ef92df0ad7a785666b6'
WITH_ASAN is ON. ASAN patch will be applied.              #### `WITH_ASAN ` is properly enabled ####

0% [Working]
0% [Connecting to archive.ubuntu.com (91.189.88.152)] [Waiting for headers]
```

### build-ceph-asan-container

#### Check TAG files

generated script
```
dir=ceph                                 #### `dir` is set to ceph ####
if [ "$dir" = "" ]; then dir=ceph ; fi
image=ceph                                 #### `image` is set to ceph ####
targets=""
if [ "$targets" != "" ]; then
  for target in $targets; do break; done # get first element
  image=$image-$target
fi
suffix=-asan                                     #### `suffix` is set to -asan ####
c="$(./tag_exists $image $dir $suffix)"
```

File: ceph/deb.sh [hunk 1]
```diff
@@ -3,12 +3,16 @@ set -eu
 
 CEPH_DIR=$(readlink -f $(dirname $0))
 
-if [ $# -ne 1 ]; then
-    echo "deb.sh VERSION"
+if [ $# -lt 1 ] || [ $# -gt 2 ]; then
```
Comment (toshipp) [inline ceph/deb.sh:6]: I think it is better to use `getopts` for optional arguments.

Comment (peng225) [reply ceph/deb.sh:6]: Is it OK to use `getopts` for both `VERSION` and `WITH_ASAN` parameters?
I think that mixing different way of getting parameters (I mean, one is using `getopts' and the other is not) results in the complex parameter parsing logic.
For example, the following command works well.
```
./deb.sh -a 16.2.7
```
On the other hand, the following one doesn't.
```
./deb.sh 16.2.7 -a 
```

So I think the following one is good.
```
./deb.sh -v 16.2.7 -a    # OK!
./deb.sh -a -v 16.2.7    # OK!
```

Could you give me a comment about it?

Comment (toshipp) [reply ceph/deb.sh:6]: The version parameter is mandatory, so I prefer `deb.sh -a 16.2.7` to `deb.sh -a -v 16.2.7`.
Many cli programs use `prog [opt...] arg...` style, I think it is unnecessary to support `deb.sh 16.2.7 -a`.

Comment (peng225) [reply ceph/deb.sh:6]: Thank you for your comment. I will adopt `prog [opt...] arg...` style.

**Comment by peng225** [comment]
Latest CI results:
### build-ceph-asan

#### Build Ceph packages with ASAN enabled

generated script
```
# set WITH_ASAN flag to ON
ceph/deb.sh -a 16.2.7
```

console output
```
WITH_ASAN is ON. ASAN patch will be applied.
```

### build-ceph-asan-container

#### Check TAG files

generated script
```
#!/bin/sh -eo pipefail
dir=ceph
if [ "$dir" = "" ]; then dir=ceph ; fi
image=ceph
targets=""
if [ "$targets" != "" ]; then
  for target in $targets; do break; done # get first element
  image=$image-$target
fi
suffix=-asan
```

**Review by toshipp — APPROVED** [review]
LGTM

**Review by llamerada-jp — APPROVED** [review]

**Pull request merged**